vespalib/src/vespa/vespalib/stllike/lrucache_map.hpp

```diff
@@ -146,12 +146,16 @@ lrucache_map<P>::erase(const iterator & it)
 }
 
 template< typename P >
-bool
+void
 lrucache_map<P>::verifyInternals()
 {
-    bool retval(true);
-    assert(_head != LinkedValueBase::npos);
-    assert(_tail != LinkedValueBase::npos);
+    if (_head == LinkedValueBase::npos || _tail == LinkedValueBase::npos) {
+        // If _either_ head or tail is invalid, they must both be.
+        assert(_head == LinkedValueBase::npos);
```

Comment:
consider using REQUIRE_EQ as a modern replacement for assert that also works with GTest and reports actual values